### ISSUE TITLE
compilation errors with CMake 3.8 [$25]

### ISSUE THREAD

**Issue by decathorpe** [elementary/switchboard-plug-pantheon-shell#51]
I get the following log messages during compilation of the latest revision(s) atop GNOME 3.24 (RC) libs:

```
(...)
/usr/bin/cc  -DGETTEXT_PACKAGE=\"pantheon-desktop-plug\" -DGNOME_DESKTOP_USE_UNSTABLE_API -I/builddir/build/BUILD/switchboard-plug-pantheon-shell-0.2.4+rev575/set-wallpaper-contract  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic   -w -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/switchboard-2.0 -I/usr/include/gee-0.8 -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libdrm -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/granite -I/usr/include/gnome-desktop-3.0 -I/usr/include/gsettings-desktop-schemas -I/usr/include/plank -I/usr/include/libbamf3 -I/usr/lib64/libbamf3/include "-pthread -DGETTEXT_PACKAGE=pantheon-desktop-plug" -o CMakeFiles/set-wallpaper.dir/set-wallpaper.c.o   -c /builddir/build/BUILD/switchboard-plug-pantheon-shell-0.2.4+rev575/set-wallpaper-contract/set-wallpaper.c
cc: error: unrecognized command line option '-pthread -DGETTEXT_PACKAGE=pantheon-desktop-plug'
(...)
```

It seems there are some misplaced `"` in the build configuration, see ` "-pthread -DGETTEXT_PACKAGE=pantheon-desktop-plug" `, which causes the error above.

**Comment by tintou** [comment]
Can you see if this fixes the build ?
https://github.com/elementary/switchboard-plug-pantheon-shell/tree/wip/tintou/fix-build-latest

**Comment by decathorpe** [comment]
No, nothing has changed with those changes.

I think the issue is in `set-wallpaper-contract/CMakeLists.txt` at lines 14/15,

```
(...)
set_target_properties (${TARGET} PROPERTIES
	COMPILE_OPTIONS "${DEPS_CFLAGS} -DGETTEXT_PACKAGE=${GETTEXT_PACKAGE}")
(...)
```

where the enclosing quotation marks seem to get included in the compilation command erroneously.

**Comment by decathorpe** [comment]
Contrary to what I initially believed to be the cause of this error, it seems to be caused by the update to CMake 3.8.

**Comment by decathorpe** [comment]
I can confirm that the linked fix (#58) indeed fixes compilation with CMake 3.8. Thanks!
